pkg/framework/plugins/defaultevictor/defaultevictor.go

```diff
@@ -0,0 +1,199 @@
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type defaultEvictorFilterArgs, got %T", args)
+	}
+	options := &FilterOptions{}
+	for _, opt := range opts {
+		opt(options)
+	}
+
+	ev := &DefaultEvictor{}
+	evictFailedBarePods := defaultEvictorArgs.EvictFailedBarePods
+	evictSystemCriticalPods := defaultEvictorArgs.EvictSystemCriticalPods
+	evictLocalStoragePods := defaultEvictorArgs.EvictLocalStoragePods
+	ignorePvcPods := defaultEvictorArgs.IgnorePvcPods
+	nodes := defaultEvictorArgs.Nodes
+	if evictFailedBarePods {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			ownerRefList := podutil.OwnerRef(pod)
+			// Enable evictFailedBarePods to evict bare pods in failed phase
+			if len(ownerRefList) == 0 && pod.Status.Phase != v1.PodFailed {
+				return fmt.Errorf("pod does not have any ownerRefs and is not in failed phase")
+			}
+			return nil
+		})
+	} else {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			ownerRefList := podutil.OwnerRef(pod)
+			if len(ownerRefList) == 0 {
+				return fmt.Errorf("pod does not have any ownerRefs")
+			}
+			return nil
+		})
+	}
+	if !evictSystemCriticalPods {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			if utils.IsCriticalPriorityPod(pod) {
+				return fmt.Errorf("pod has system critical priority")
+			}
+			return nil
+		})
+
+		if options.Priority != nil {
+			ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+				if IsPodEvictableBasedOnPriority(pod, *options.Priority) {
+					return nil
+				}
+				return fmt.Errorf("pod has higher priority than specified priority class threshold")
+			})
+		}
+	}
+	if !evictLocalStoragePods {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			if utils.IsPodWithLocalStorage(pod) {
+				return fmt.Errorf("pod has local storage and descheduler is not configured with evictLocalStoragePods")
+			}
+			return nil
+		})
+	}
+	if ignorePvcPods {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			if utils.IsPodWithPVC(pod) {
+				return fmt.Errorf("pod has a PVC and descheduler is configured to ignore PVC pods")
+			}
+			return nil
+		})
+	}
+	if options.NodeFit {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			if !nodeutil.PodFitsAnyOtherNode(nodeIndexer, pod, nodes) {
+				return fmt.Errorf("pod does not fit on any other node because of nodeSelector(s), Taint(s), or nodes marked as unschedulable")
+			}
+			return nil
+		})
+	}
+	if options.LabelSelector != nil && !options.LabelSelector.Empty() {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			if !options.LabelSelector.Matches(labels.Set(pod.Labels)) {
+				return fmt.Errorf("pod labels do not match the labelSelector filter in the policy parameter")
+			}
+			return nil
+		})
+	}
+
+	return ev, nil
+}
+
+// Name retrieves the plugin name
+func (d *DefaultEvictor) Name() string {
```

Comment:
Is this and the `PluginName` const actually needed?

Reply:
Do you mean we should only add this one if we have other Evictor options added in the codebase?

Reply:
The `PluginName` const will be needed once the plugin registry is introduced as a key in the corresponding map.

Reply:
`Name()` is needed so all plugins implement `Plugin` interface.

Reply:
:+1: